Update-HostsFile.ps1

```diff
@@ -0,0 +1,170 @@
+
+    .LINK
+    https://github.com/StevenBlack/hosts
+    #>
+    [cmdletBinding(DefaultParameterSetName='UnifiedOnly')]
+    param
+    (
+        [Switch]
+        [Parameter(ParameterSetName='UnifiedOnly',Mandatory=$false)]
+        $UnifiedOnly,
+        [Switch]
+        [Parameter(ParameterSetName='Extension',Mandatory=$false)]
+        $Fakenews,
+        [Switch]
+        [Parameter(ParameterSetName='Extension',Mandatory=$false)]
+        $Gambling,
+        [Switch]
+        [Parameter(ParameterSetName='Extension',Mandatory=$false)]
+        $Social,
+        [Switch]
+        [Parameter(ParameterSetName='Extension',Mandatory=$false)]
+        $Porn,
+        [Parameter(ParameterSetName='All',Mandatory=$false)]
+        [Switch]
+        $All,
+        [Parameter()]
+        [ValidateScript({ Test-Path -Path $_ -PathType Leaf})]
+        [string]
+        $WhitelistFilePath,
+        [ValidateScript({ Test-Path -Path $_ -PathType Leaf})]
+        [string]
+        $BlackListFilePath,
+        [switch]
+        $BackupOldHostsFile
+    )
+    if(-not ([Security.Principal.WindowsPrincipal][Security.Principal.WindowsIdentity]::GetCurrent()).IsInRole([Security.Principal.WindowsBuiltInRole]::Administrator))
+    {
+        throw 'This script requires local administrator privilege'
+    }
+
+    $BaseUri = 'https://raw.githubusercontent.com/StevenBlack/hosts/master'
+    if($PSBoundParameters.ContainsKey('All'))
+    {
+        $null = $PSBoundParameters.Remove('All')
+        $null = $PSBoundParameters.Add('Fakenews',$true)
+        $null = $PSBoundParameters.Add('Gambling',$true)
+        $null = $PSBoundParameters.Add('Social',$true)
+        $null = $PSBoundParameters.Add('Porn',$true)
+    }
+    
+    if($PSBoundParameters.Keys.Count -eq 0 -or $PSBoundParameters.ContainsKey('UnifiedOnly'))
+    {
+        Write-Verbose 'Processing UnifiedOnly Hosts file'
+        $downloadUri = $BaseUri + '/hosts'
+    }
+    elseif($PSBoundParameters.Keys.Count -ge 1 -and -not $PSBoundParameters.ContainsKey('All'))
+    {
+        
+        $downloadUri = $BaseUri + '/alternates/' + (( $PSBoundParameters.Keys|Where-Object {$_ -notmatch 'File'}|ForEach-Object {$_.ToLower()} | Sort-Object ) -join '-') + '/hosts'
+        Write-Verbose "Using $downloadUri"
+    }
+    try{
+    $webResponse = Invoke-WebRequest -Uri $downloadUri -UseBasicParsing 
+    }
+    catch
+    {
+      throw $_
+    }
+    
+    $hostsContent = $webResponse.Content
+    $hostsArray = New-Object System.Collections.ArrayList  
+    $null = $hostsArray.AddRange(($hostsContent -split '\n'))
+    if(-not [string]::IsNullOrEmpty($WhitelistFilePath) )
+    {
+        
+        $whiteListDomains = Get-Content -Path $WhitelistFilePath
+        $hostsToWhiteList = New-Object System.Collections.ArrayList
+        foreach($hostsEntry in $hostsArray)
+        {
+            if($whiteListDomains.Contains($hostsEntry.Split('\s')[1]))

```

Comment:
IINM, `.Contains` doesn't do regex matching?  I ask because exclusions on the Python-side can be regex.

Reply:
Thanks @gfyoung.